WordPress/src/main/java/org/wordpress/android/util/extensions/StringExtensions.kt

```diff
@@ -17,3 +20,26 @@ import java.util.Locale
 fun String.capitalizeWithLocaleWithoutLint(locale: Locale): String {
     return this.capitalize(locale)
 }
+
+/**
+ * Converts digits to Western Arabic -- Workaround for an issue in Android that shows Eastern Arabic numerals.
+ * There is an issue in Google's bug tracker for this: [SO Answer](https://stackoverflow.com/a/34612487/4129245).
+ * The returned String uses the default Locale.
+ * @return a String with numerals in Western Arabic format persisting spans if available.
+ */
+fun CharSequence.enforceWesternArabicNumerals(): CharSequence {
+    val textWithArabicNumerals = this
+            // Replace Eastern Arabic numerals
+            .replace(Regex("[٠-٩]")) { match -> (match.value.single() - '٠').toString() }
+            // Replace Persian/Urdu numerals
+            .replace(Regex("[۰-۹]")) { match -> (match.value.single() - '۰').toString() }
+
+    // Restore spans if text is an instance of Spanned
+    if (this is Spanned) {
+        val spannableText = SpannableString(textWithArabicNumerals)
+        TextUtils.copySpansFrom(this, 0, this.length, null, spannableText, 0)
+        return spannableText
+    }
```

Comment:
I have no idea nor experience in how Android and `SpannableString` works, but if I'm assuming correctly that this is a class used to represent styled strings (bold, italics, etc), and if I'm understanding correctly that your `match.value.single() - '٠'` might _remove_ that character from the original strings:

 - I'm never super confident in manipulating strings in other locales by making assumptions about the meaning of individual characters. Both Unicode and Localization are quite complex topics with lots of rules and edge cases (see various "falsehoods programmers believe about […]" articles on the web), so it's usually not as easy as what you might assume at first, to cover all edge cases.
 - If that code indeed removes `'٠'` characters from the original text, that means the length of the string might changes. As such, not only will spans have to be shifted accordingly to account for this character count shift, but also, with the code as is, `this.length` will be out of bounds of the length of the shorter `spannableText`. So I'm guessing it could create a crash as a result
 - Speaking of, even if the length didn't change, are we sure that all spans make sense to copy, including, e.g. spans changing the text direction (RTL-markers like U+200F that polyglots might use to enforce text direction, etc) and similar? I think so because I'm hoping Arabic and Persian/Urdu polyglots will assume the `%d` in the originals will be western numerals and will thus use the text direction markers accordingly in their translation if needed… but it's worth thinking about anyway.

In the end my main concern is (1) wrong assumptions about Arabic and Urdu text (2) risk of OOB crash if text length changes, and especially a crash because of this would be quite bad (and easy to miss as Arabic and Persian are not the primary languages we test the app often with… except the L10n smoke-testing we do every second Wednesday, but we don't always cover all the screens during that testing)

Reply:
Oh I see, it's just a difference in the ASCII codes (well, the Unicode Code Point, since we're not dealing with ASCII but Unicode here), gotcha 👍

Reply:
> Oh I see, it's just a difference in the ASCII codes (well, the Unicode Code Points, since we're not dealing with ASCII but Unicode here), gotcha 👍

Correct 👍 , I forgot to link [this page](https://www.cs.cmu.edu/~pattis/15-1XX/common/handouts/ascii.html) that explains it nicely.